src/librbd/ExclusiveLock.cc

```diff
@@ -37,20 +37,35 @@ ExclusiveLock<I>::ExclusiveLock(I &image_ctx)
 }
 
 template <typename I>
-bool ExclusiveLock<I>::accept_requests(int *ret_val) const {
+bool ExclusiveLock<I>::accept_requests() const {
   std::lock_guard locker{ML<I>::m_lock};
 
   bool accept_requests = (!ML<I>::is_state_shutdown() &&
                           ML<I>::is_state_locked() &&
                           m_request_blocked_count == 0);
-  if (ret_val != nullptr) {
-    *ret_val = m_request_blocked_ret_val;
-  }
 
   ldout(m_image_ctx.cct, 20) << "=" << accept_requests << dendl;
   return accept_requests;
 }
 
+template <typename I>
+bool ExclusiveLock<I>::accept_request(OperationRequestType request_type,
+                                      int *ret_val) const {
+  std::lock_guard locker{ML<I>::m_lock};
+
+  auto policy = m_image_ctx.get_exclusive_lock_policy();
+  bool accept_request = (!ML<I>::is_state_shutdown() &&
+                         ML<I>::is_state_locked() &&
+                         (m_request_blocked_count == 0 ||
+                          policy->accept_blocked_request(request_type)));
+  *ret_val = m_request_blocked_ret_val;
```

Comment:
Nit: (for clarity) probably don't want to return an error code if accepts are being accepted.